docs/_guide/styles.md

```diff
@@ -162,327 +122,653 @@ class MyElement extends LitElement {
+
+For more information on these and other lit-html directives, see the documentation on [lit-html built-in directives](https://lit-html.polymer-project.org/guide/template-reference#built-in-directives).
+
+To use `styleMap` and/or `classMap`:
+
+1.  Import `classMap` and/or `styleMap`:
+
+    ```js
+    import { classMap } from 'lit-html/directives/class-map';
+    import { styleMap } from 'lit-html/directives/style-map';
+    ```
+
+2.  Use `classMap` and/or `styleMap` in your element template:
+
+    ```js
+    constructor() {
+      super();
+      this.classes = { mydiv: true, someclass: true };
+      this.styles = { color: 'green', fontFamily: 'Roboto' };
+    }
+    render() {
+      return html`
+        <div class=${classMap(this.classes)} style=${styleMap(this.styles)}>
+          Some content
+        </div>
+      `;
+    }
+    ```
+
+{% include project.html folder="style/maps" openFile="my-element.js" %}
+
+### classMap syntax {#classmap}
+
+`classMap` applies a set of classes to an HTML element:
+
+```html
+<div class=${classMap({alert:true,info:true})}>Content.</div>
+<!-- Equivalent: <div class="alert info">Content.</div> -->
+```
+
+{% include project.html folder="style/classmap" openFile="my-element.js" %}
+
+### styleMap syntax {#stylemap}
+
+`styleMap` applies a set of CSS rules to an HTML element:
 
 ```html
-  <style>
-    div { font-family: Roboto; }
-  </style>
-  <div>
-    <my-element>Will use Roboto font</my-element>
-  </div>
+<button style=${styleMap({
+  backgroundColor: 'blue',
+  border: '1px solid black'
+})}>A button</button>
+
+<!-- Equivalent: 
+  <button style="
+    background-color:blue;
+    border: 1px solid black;
+  ">A button</button>
+-->
+```
+
+{% include project.html folder="style/stylemap" openFile="my-button.js" %}
+
+To refer to hyphenated properties such as `font-family`, use the camelCase equivalent (`fontFamily`) or place the hyphenated property name in quotes (`'font-family'`). 
+
+To refer to custom CSS properties such as `--custom-color`, place the whole property name in quotes (`'--custom-color'`).
+
+|**Inline style or CSS**|**styleMap equivalent**|
+|----|----|
+| `background-color: blue;` <br/> | `backgroundColor: 'blue'` <br/><br/> or <br/><br/>`'background-color': 'blue'`|
+| `font-family: Roboto, Arial, sans-serif;` <br/> | `fontFamily: 'Roboto, Arial, sans-serif'` <br/><br/> or <br/><br/>`'font-family': 'Roboto, Arial, sans-serif'`|
+|`--custom-color: #FFFABC;`|`'--custom-color': '#FFFABC;'`|
+|`--otherCustomColor: #FFFABC;`|`'--otherCustomColor': '#FFFABC;'`|
+|`color: var(--customprop, blue);`|`color: 'var(--customprop, blue)'`|
+
+**Examples**
+
+Inline style syntax:
+
+```html 
+<div style="
+  background-color:blue;
+  font-family:Roboto;
+  --custom-color:#e26dd2;
+  --otherCustomColor:#77e26d;">
+</div>
 ```
 
-{% include project.html folder="style/inherited2" openFile="my-element.js" %}
+Equivalent CSS syntax:
```

Comment:
Seems a little repetitive too. I think you could maybe drop this "Examples" section. There are already a number of examples explaining the same thing above, plus text explaining it, plus a table, so this all feels a little over-explained to me.

Really the only interesting thing to me about `styleMap` is that `dash-case` gets turned into `camelCase` -- quoting property names is just how JavaScript objects works.

I think I basically just want to see *one* example of using `styleMap` that shows all the possible forms:

```
<button style=${styleMap({
  border: '1px solid black',

  'background-color': 'blue',
  backgroundColor: 'blue', // Equivalent to above

  '--custom-color': 'red',
  '--customColor': 'red', // Equivalent to above
})}>A button</button>